File: Algorithms/487. Max Consecutive Ones II/javascript/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function (nums) {
    let prev = -1,
        curr = 0,
        maxLength = 0;

    for (const num of nums) {
        if (num === 0) {
            [prev, curr] = [curr, 0];
        } else {
            curr++;
        }

        maxLength = Math.max(maxLength, curr + prev + 1);
    }

    return maxLength;
};

File: Algorithms/487. Max Consecutive Ones II/golang/solution.go
func findMaxConsecutiveOnes(nums []int) int {
	var prev, curr, maxLen int = -1, 0, 0

	for _, num := range nums {
		if num == 0 {
			prev, curr = curr, 0
		} else {
			curr++
		}

		if prev+1+curr > maxLen {
			maxLen = prev + 1 + curr
		}
	}

	return maxLen
}